### PR TITLE
More fix mail relay

### PR DESCRIPTION
**Pull request by JedMeister**
Fix the Mail relay form function, plus improved error handling. Also allow unauthenticated SMTP relay.

Closes https://github.com/turnkeylinux/tracker/issues/844

### CODE REVIEW

**Comment by JedMeister** [comment]
@OnGle - I've merged this so I can build and test properly, but please still cast yoru eye over it and provide any thoughts/feedback. Cheers.

File: plugins.d/Mail_Relaying/mail_relay.py [hunk 1]
```diff
@@ -23,34 +25,44 @@
 
 
 def testsettings(host, port, login, password):
-    host = host.encode('utf-8')
+
+    encoding = sys.stdin.encoding
+
+    # login username and password must be string
+    def bytes2string(string):
+        if type(string) is bytes:
```
Comment (OnGle) [inline plugins.d/Mail_Relaying/mail_relay.py:33]: Generally ``isinstance(string, bytes)`` would be prefered to ``type(string) == bytes`` or ``type(string) is bytes`` as it handles inheritance, although it **probably** doesn't really matter here

File: plugins.d/Mail_Relaying/mail_relay.py [hunk 2]
```diff
@@ -83,27 +101,51 @@ def run():
...
-                ('Login', login, field_width, field_limit),
-                ('Password', password, field_width, field_limit)
+                ('Host', 1, 0, host, 1, 10, field_width, field_limit),
+                ('Port', 2, 0, port, 2, 10, field_width, field_limit),
+                ('Login', 3, 0, login, 3, 10, field_width, field_limit),
+                ('Password', 4, 0, password, 4, 10, field_width, field_limit)
             ]
 
             retcode, values = console.form(TITLE, FORMNOTE, fields)
+            host, port, login, password = tuple(values)
```
Comment (OnGle) [inline plugins.d/Mail_Relaying/mail_relay.py:111]: does ``host, port, login, password = values`` not work here? Is there a reason to casting this as a tuple?

File: plugins.d/Mail_Relaying/mail_relay.py (hunk 1 above)
Comment (JedMeister) [reply plugins.d/Mail_Relaying/mail_relay.py:33]: Hmm. Ok thanks... I'll leave it be for now anyway.

File: plugins.d/Mail_Relaying/mail_relay.py (hunk 2 above)
Comment (JedMeister) [reply plugins.d/Mail_Relaying/mail_relay.py:111]: Nope, just that's what it was before. I only changed the order (and perhaps the indent level too?). The line itself was already there...

File: plugins.d/Mail_Relaying/mail_relay.py [hunk 3]
```diff
@@ -83,27 +101,51 @@ def run():
...
 
             if retcode is not 'ok':
                 console.msgbox(TITLE,
                                'You have cancelled the configuration process.'
                                ' No relaying of mail will be performed.')
                 return
 
-            host, port, login, password = tuple(values)
```
Comment (JedMeister) [inline plugins.d/Mail_Relaying/mail_relay.py:100]: @OnGle - here's the previous line, albeit in a slightly different place.

Comment (JedMeister) [reply plugins.d/Mail_Relaying/mail_relay.py:100]: FWIW it was there right from the start: https://github.com/turnkeylinux/confconsole/commit/e87fda7deb2a19329578d009754af2fe93287ebe

Comment (OnGle) [reply plugins.d/Mail_Relaying/mail_relay.py:100]: Oh yep fair enough, still odd to me but not by anyway :p

**Pull request merged**